File: arrays/Count Number of Pairs With Absolute Difference K.cpp
// TC:O(n^2) SC:O(1)
class Solution {
public:
    int countKDifference(vector<int>& nums, int k) {
        int cnt=0;
        for(int i=0;i<nums.size();i++){
            for(int j=i+1;j<nums.size();j++){
                if(abs(nums[i]-nums[j]) ==k)cnt++;
            }
        }
        return cnt;

    }
};
// TC:O(n) SC:O(n)
class Solution {
public:
    int countKDifference(vector<int>& nums, int k) {
        unordered_map<int,int> mpp;
        int cnt = 0;
        for (int n : nums) {
           cnt+=mpp[n-k];
           cnt+=mpp[n+k];
           mpp[n]++;
        }
        return cnt;
    }
};

File: arrays/Python/Count Number of Pairs With Absolute Difference K.py
// TC:O(n^2) SC:O(1)
class Solution:
    def countKDifference(self, nums, k):
        cnt = 0
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if abs(nums[i] - nums[j]) == k:
                    cnt += 1
        return cnt

// TC:O(n) SC:O(n)
class Solution:
    def countKDifference(self, nums, k):
        mpp = {}
        cnt = 0
        for n in nums:
            cnt += mpp.get(n - k, 0)
            cnt += mpp.get(n + k, 0)
            mpp[n] = mpp.get(n, 0) + 1
        return cnt
